### ISSUE TITLE
Normalizing loss over entire dataset vs normalizing over number of batches.

### ISSUE THREAD

**Issue by fedeotto** [unnir/cVAE#2]
I've noticed that when you display the training loss:

```
    print('====> Epoch: {} Average loss: {:.4f}'.format(
          epoch, train_loss / len(train_loader.dataset)))
```

you normalize that by the length of the _entire_ dataset. Couldn't you instead divide it by the number of batches you passing for that epoch, in order to have an idea of the average loss based on batches?

**Comment by unnir** [comment]
this is a correct implementation, since we accumulate the epoch loss in the train_loss variable. 

For the batch loss, we do normalize it by the batch size: 

```python
        if batch_idx % 20 == 0:
            print('Train Epoch: {} [{}/{} ({:.0f}%)]\tLoss: {:.6f}'.format(
                epoch, batch_idx * len(data), len(train_loader.dataset),
                100. * batch_idx / len(train_loader),
                loss.item() / len(data)))
```
